application: serial_lte_modem: new nRF Cloud service (#11573)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/NRFCLOUD_AT_commands.rst

.. _SLM_AT_NRFCLOUD:

nRF Cloud AT commands
*********************

.. contents::
   :local:
   :depth: 2

The following commands list contains nRF Cloud-related AT commands.

nRF Cloud access
================

The ``#XNRFCLOUD`` command controls the access to the nRF Cloud service.

.. note::

>    To use ``#XNRFCLOUD``, you must first provision the device to nRF Cloud, using the UUID from the modem firmware as device ID.

Author: Added

Example
~~~~~~~

::

  AT%XSYSTEMMODE=1,0,0,0

  OK
  AT+CFUN=1

  OK
  AT#XNRFCLOUD=1

  OK
  #XNRFCLOUD: 1,0
  AT#XCELLPOS=1

  OK

  #XCELLPOS: 0,35.455833,139.626111,1094

  AT%NCELLMEAS

  OK

  %NCELLMEAS: 0,"0199F10A","44020","107E",65535,3750,5,49,27,107504,3750,251,33,4,0,475,107,26,14,25,475,58,26,17,25,475,277,24,9,25,475,51,18,1,25

  AT#XCELLPOS=2

  OK

  #XCELLPOS: 1,35.534999,139.722362,1801
  AT#XCELLPOS=0

  OK

Read command
------------

The read command allows you to check the cellular location service status.

Syntax
~~~~~~

::

   #XCELLPOS?

Response syntax
~~~~~~~~~~~~~~~

::

   #XCELLPOS: <cellpos_status>

* The ``<cellpos_status>`` value is an integer.
  When it returns the value of ``1``, it means that the cellular location service is started.

Example
~~~~~~~

::

  AT#XCELLPOS?

  #XCELLPOS: 1

  OK

Test command
------------

The test command tests the existence of the command and provides information about the type of its subparameters.

Syntax
~~~~~~

::

   #XCELLPOS=?

Example
~~~~~~~

::

  AT#XCELLPOS=?

  #XCELLPOS: (0,1,2)

  OK

nRF Cloud Wi-Fi location
========================

> The ``#XWIFIPOS`` command runs the nRF Cloud Wi-Fi location service for location information.

Reviewer: ```suggestion
The ``#XWIFIPOS`` command runs the nRF Cloud Wi-Fi location service to receive location information.
```

.. note::
   To use ``#XWIFIPOS``, the following preconditions apply:

   * You must define :ref:`CONFIG_SLM_NRF_CLOUD <CONFIG_SLM_NRF_CLOUD>` and :ref:`CONFIG_NRF_CLOUD_LOCATION <CONFIG_NRF_CLOUD_LOCATION>`.
   * You must have access to nRF Cloud through the LTE network.

Set command
-----------

The set command allows you to start and stop the nRF Cloud Wi-Fi location service.

Syntax
~~~~~~

::

   #XWIFIPOS=<op>[,<ssid0>,<mac0>[,<ssid1>,<mac1>[...]]]

The ``<op>`` parameter accepts the following integer values:

* ``0`` - Stop Wi-Fi location.
* ``1`` - Start Wi-Fi location.

* The ``<ssidX>`` parameter is a string.
  It indicates the SSID of the Wi-Fi access point.

* The ``<macX>`` parameter is a string.

>   It indicates the MAC address string of the Wi-Fi access point.
>   The string should be formatted as "%02x:%02x:%02x:%02x:%02x:%02x".

Reviewer: ```suggestion
  It indicates the MAC address of the Wi-Fi access point, and  should be formatted as ``%02x:%02x:%02x:%02x:%02x:%02x``.
```

> The command accepts ``<ssidX>`` and ``<macX>`` of up to 5 access points.

Reviewer: ```suggestion
The command accepts the ``<ssidX>`` and ``<macX>`` values of up to 5 access points.
```
